client/signup/main.jsx

```diff
@@ -286,20 +293,22 @@ class Signup extends React.Component {
 
 			fulfilledSteps.push( siteTopicStepName );
 
-			analytics.tracks.recordEvent( 'calypso_signup_actions_exclude_step', {
-				step: siteTopicStepName,
-				value: vertical,
-			} );
+			this.recordExcludeStepEvent( siteTopicStepName, vertical );
 		}
 
 		//`site_type` query parameter
 		const siteTypeValue = getSiteTypePropertyValue( 'slug', siteType, 'slug' );
 		if ( 'undefined' !== typeof siteTypeValue ) {
 			debug( 'From query string: site_type = %s', siteType );
 			debug( 'Site type value = %s', siteTypeValue );
-			this.props.setSiteType( siteTypeValue );
-			// TODO:
-			// exlude the site type step if it is fulfilled here.
+
+			const siteTypeStepName = 'site-type';
+
+			this.props.submitSiteType( siteTypeValue );
+
+			fulfilledSteps.push( siteTypeStepName );
```

Comment:
Are fulfilled steps different to other kinds of state for a step?

Reply:
Sorry that I have overlooked your inquiry for so long. The underlying states are the same for those pre-fulfilled steps, just being excluded beforehand. Eventually when we get to a general solution, I believe it will change.